src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const adder = (value) => {
+    if (value === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
+
+    sum += value;
+
+    return adder;
+  };
```

Comment:
Your implementation of the `adder` function correctly handles both the accumulation of values and the reset of the sum when called without arguments. However, you should consider the potential issue of closure over the `sum` variable. Each call to `makeInfinityAdder` will create a new closure over `sum`, which is good for encapsulation, but if `makeInfinityAdder` is called frequently, it might lead to a memory leak as each closure's state is retained in memory. This is not necessarily a bug, but something to be aware of in applications where `makeInfinityAdder` would be called many times.